### PR TITLE
PMM-7380 DB Model

### PR DESCRIPTION
**Pull request by Dasio**
[PMM-7380](https://jira.percona.com/browse/PMM-7380)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/0

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: models/backup_model.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+//go:generate reform
+
+// BackupLocationType represents BackupLocation type as stored in database.
+type BackupLocationType string
+
+// BackupLocation types
+const (
+	S3BackupLocationType NodeType = "s3"
```
Comment (artemgavrilov) [inline models/backup_model.go:32]: Seems that type here should be `BackupLocationType`

```suggestion
	S3BackupLocationType BackupLocationType = "s3"
```

File: models/backup_model.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	S3BackupLocationType BackupLocationType = "s3"
+	FSBackupLocationType BackupLocationType = "fs"
+)
+
+// BackupLocation represents destination for backup.
+//reform:backup_locations
+type BackupLocation struct {
+	ID          string             `reform:"id,pk"`
+	Name        string             `reform:"name"`
+	Description *string            `reform:"description"`
```
Comment (artemgavrilov) [inline models/backup_model.go:41]: Why it's a pointer?

File: models/backup_model.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	Type        BackupLocationType `reform:"type"`
+	Endpoint    *string            `reform:"endpoint"`
+	AccessKey   *string            `reform:"access_key"`
+	SecretKey   *string            `reform:"secret_key"`
+	Path        *string            `reform:"path"`
```
Comment (artemgavrilov) [inline models/backup_model.go:46]: I think it better to create separate config models and store them as jsons. Same as https://github.com/percona/pmm-managed/blob/787c43114f2bcdcce597e1081cceb2f9ea1f8b98/models/channel.go#L46-L49

File: models/backup_model.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+	CreatedAt time.Time `reform:"created_at"`
+	UpdatedAt time.Time `reform:"updated_at"`
```
Comment (atymchuk) [inline models/backup_model.go:49]: @artemgavrilov We will have an audit requirement very soon, maybe we should provision for it now?

File: models/backup_model.go (hunk 2 above)
Comment (Dasio) [reply models/backup_model.go:41]: I thought when it's option then it's pointer.
https://github.com/percona/pmm-managed/blob/PMM-2.0/models/agent_model.go#L98

File: models/backup_model.go (hunk 4 above)
Comment (artemgavrilov) [reply models/backup_model.go:49]: Yep, I think it's a good idea 👍

Comment (atymchuk) [reply models/backup_model.go:48]: I take it back - this should go to the Audit component, which will be designed later down the road.

Comment (Dasio) [reply models/backup_model.go:48]: Auditing is out of scope of this issue/feature.
Would be handled separetely.

File: models/backup_model.go (hunk 2 above)
Comment (artemgavrilov) [reply models/backup_model.go:42]: Let's make this filed `string`, we don't need to distinguish between nil and empty sting here. And it's easier to use simple strings.

**Review by artemgavrilov — APPROVED** [review]

**Review by oter — APPROVED** [review]

**Pull request merged**